raidprotect/src/event/message/handle.rs

```diff
@@ -14,5 +16,64 @@ pub async fn handle_message(message: Message, state: Arc<ClusterState>) {
     let parsed = parse_message(&message);
     state.redis().set(&parsed).await.ok();
 
+    if is_old_command(&message.content) {
+        let message = message.clone();
+        let state = state.clone();
+
+        tokio::spawn(warn_old_command(message, state));
+    }
+
     info!("received message: {}", message.content) // Debug util real implementation
 }
+
+async fn warn_old_command(message: Message, state: Arc<ClusterState>) {
+    let lang = message
+        .author
+        .locale
+        .map(|locale| Lang::from(locale.as_str()))
+        .unwrap_or_else(|| Lang::fallback());
+    let embed = EmbedBuilder::new()
+        .title(lang.warning_deprecated_command_title())
+        .description(lang.warning_deprecated_command_description())
+        .color(COLOR_TRANSPARENT)
+        .build();
+
+    match state
+        .http()
+        .create_message(message.channel_id)
+        .embeds(&[embed])
+    {
+        Ok(msg) => {
+            let msg = msg.reply(message.id);
+
+            if let Err(error) = msg.exec().await {
+                error!(error = ?error, "failed to warn user about question mark commands deprecation");
+            }
+        }
+        _ => error!("failed to create embed"),
+    }
+}
+
+fn is_old_command(content: &str) -> bool {
+    const OLD_COMMANDS: [&str; 11] = [
+        "?kick",
+        "?userinfo",
+        "?lock",
+        "?unlock",
+        "?raidmode",
+        "?settings",
+        "?captcha",
+        "?ban",
+        "?invite",
+        "?clear",
+        "?stats",
+    ];
+
+    for cmd in OLD_COMMANDS {
+        if content.starts_with(cmd) {
+            return true;
+        }
```

Comment:
Why don't using something like that? :
`OLD_COMMANDS.iter().any(|s| text.starts_with(s));`

[Rust Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=2a9524c0f182cd02dcd80b9138efaeab)

```suggestion
    return OLD_COMMANDS.iter().any(|s| text.starts_with(s));
```

Reply:
It's a real question, not a change request

Reply:
Thanks, honestly I thought about something like that but I didn't search for it.

Reply:
I merged upstream and took your suggestion into account. Thanks